pkg/cloudproviders/k8s/k8s.go

```diff
@@ -0,0 +1,348 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/atlassian/gostatsd"
+	"github.com/atlassian/gostatsd/pkg/util"
+
+	"github.com/sirupsen/logrus"
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	core_v1inf "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+// NeverMatchRegex is a regex string that will never match anything
+const NeverMatchRegex = "k^"
+
+var (
+	// DefaultAnnotationTagRegex is the default ParamAnnotationTagRegex. Everything beginning with a standard
+	// gostatsd string is included by default. The default sets the tag name as the part after the "/", for example
+	// gostatsd.atlassian.com/tag1 would return "tag1" as the tag name after matching.
+	DefaultAnnotationTagRegex = "^" + regexp.QuoteMeta(AnnotationPrefix) + DefaultTagCaptureRegex
+	// DefaultLabelTagRegex is the default ParamLabelTagRegex. Every label is ignored by default.
+	DefaultLabelTagRegex = NeverMatchRegex // regex will never match anything
```

Comment:
I wonder if we should use an empty string as such flag? Has two benefits:
- Less ugly than `k^`
- Simple string check (or `nil` check - see below) is likely much faster than running a regex, even a simple one.
- The check can guard the whole loop on labels - no need to try to match each label if we already know none will match. Big performance gain.

Can be passed as a `nil` pointer to `Regexp` to indicate that it's `off`.

WDYT?

Reply:
Makes sense, I'll do that.